### PR TITLE
DEVX-1912 cloud-etl Update ksqlDB create table statement

### PR DESCRIPTION
**Pull request by rspurgeon**
to restore inclusion of eventSourceIP in the materialized value record & schema

### CODE REVIEW

**Comment by rspurgeon** [comment]
I have a change coming to docs/index.rst to capture the expected result of the new ksqlDB statements

File: cloud-etl/statements.sql [hunk 1]
```diff
@@ -1,3 +1,3 @@
 CREATE STREAM eventlogs (eventSourceIP varchar, eventAction varchar, Result varchar, eventDuration bigint) WITH (kafka_topic='eventlogs', value_format='JSON');
 CREATE TABLE count_per_source WITH (KAFKA_TOPIC='COUNT_PER_SOURCE', PARTITIONS=6) AS SELECT eventSourceIP, count(*) as COUNT FROM eventlogs GROUP BY eventSourceIP EMIT CHANGES;
-CREATE TABLE sum_per_source WITH (KAFKA_TOPIC='SUM_PER_SOURCE', PARTITIONS=6, VALUE_FORMAT='AVRO') AS SELECT eventSourceIP, sum(eventDuration) as SUM FROM eventlogs WHERE Result='Pass' GROUP BY eventSourceIP EMIT CHANGES;
+CREATE TABLE sum_per_source WITH (KAFKA_TOPIC='SUM_PER_SOURCE', PARTITIONS=6, VALUE_FORMAT='AVRO') AS SELECT EVENTLOGS.EVENTSOURCEIP ROWKEY, AS_VALUE(EVENTLOGS.EVENTSOURCEIP) EVENTSOURCEIP, SUM(EVENTLOGS.EVENTDURATION) SUM FROM EVENTLOGS EVENTLOGS WHERE (EVENTLOGS.RESULT = 'Pass') GROUP BY EVENTLOGS.EVENTSOURCEIP EMIT CHANGES;
```
Comment (ybyzek) [inline cloud-etl/statements.sql]: @rspurgeon why is it necessary to prepend `EVENTLOGS.`?  It's inconsistent with above, and makes it a bit harder to read.  Is `AS_VALUE` not sufficient?

Comment (rspurgeon) [reply cloud-etl/statements.sql]: it's the strict version of the command copied from C3 (in the running streams view).  I'll modify it to be the more 'human friendly' version and push it along w/ the change to docs i'm preparing

Comment (rspurgeon) [reply cloud-etl/statements.sql]: @ybyzek please see updated version.  Thank you

File: cloud-etl/docs/index.rst [hunk 2]
```diff
@@ -354,25 +354,40 @@ ksqlDB
...
 
    .. code:: bash
-
       [
         {
           "@type": "queries",
           "statementText": "SHOW QUERIES;",
           "queries": [
             {
+              "queryString": "CREATE TABLE COUNT_PER_SOURCE WITH (KAFKA_TOPIC='COUNT_PER_SOURCE', PARTITIONS=6, REPLICAS=3) AS SELECT\n  EVENTLOGS.EVENTSOURCEIP EVENTSOURCEIP,\n  COUNT(*) COUNT\nFROM EVENTLOGS EVENTLOGS\nGROUP BY EVENTLOGS.EVENTSOURCEIP\nEMIT CHANGES;",
```
Comment (ybyzek) [inline cloud-etl/docs/index.rst:363]: @rspurgeon should this queryString exactly match the query in `statements.sql`?  Same question for below.

File: cloud-etl/statements.sql [hunk 3]
```diff
@@ -1,3 +1,3 @@
 CREATE STREAM eventlogs (eventSourceIP varchar, eventAction varchar, Result varchar, eventDuration bigint) WITH (kafka_topic='eventlogs', value_format='JSON');
 CREATE TABLE count_per_source WITH (KAFKA_TOPIC='COUNT_PER_SOURCE', PARTITIONS=6) AS SELECT eventSourceIP, count(*) as COUNT FROM eventlogs GROUP BY eventSourceIP EMIT CHANGES;
-CREATE TABLE sum_per_source WITH (KAFKA_TOPIC='SUM_PER_SOURCE', PARTITIONS=6, VALUE_FORMAT='AVRO') AS SELECT eventSourceIP, sum(eventDuration) as SUM FROM eventlogs WHERE Result='Pass' GROUP BY eventSourceIP EMIT CHANGES;
+CREATE TABLE sum_per_source WITH (KAFKA_TOPIC='SUM_PER_SOURCE', PARTITIONS=6, VALUE_FORMAT='AVRO') AS SELECT eventSourceIP ROWKEY, as_value(eventSourceIP) EVENTSOURCEIP, SUM(EVENTDURATION) SUM FROM eventlogs WHERE (RESULT = 'Pass') GROUP BY EVENTSOURCEIP EMIT CHANGES;
```
Comment (ybyzek) [inline cloud-etl/statements.sql]: For my own education, why is `AS` not required?  i.e., how is it that `sum(eventDuration) as SUM` changed to `SUM(EVENTDURATION) SUM`

Comment (rspurgeon) [reply cloud-etl/statements.sql]: I believe the `AS` can be implied in most SQL dialects, but we should be consistent ( I didn't notice they were missing here, thank you).  I'll add `AS` across the board and make sure the casing consistent as well.

File: cloud-etl/docs/index.rst (hunk 2 above)
Comment (rspurgeon) [reply cloud-etl/docs/index.rst:363]: This is copied directly from the HTTP returned value from ksqlDB server.  IIUC, ksqlDB resolves the query you submit into something more explicit, and the value you get here won't necessarily match what you submit.  I double checked this with the updated statement in the latest commit as well as in the ksqlDB editor in Confluent Cloud.

File: cloud-etl/statements.sql (hunk 3 above)
Comment (rspurgeon) [reply cloud-etl/statements.sql]: Updated statements.sql, please check me.  thank you

**Review by ybyzek — APPROVED** [review]
LGTM.  Thanks for fixing and validating!

File: cloud-etl/docs/index.rst [hunk 4]
```diff
@@ -354,25 +354,40 @@ ksqlDB
...
+              ],
+              "id": "CTAS_COUNT_PER_SOURCE_0",
+              "statusCount": {
+                "RUNNING": 1
+              },
+              "queryType": "PERSISTENT",
+              "state": "RUNNING"
             },
             {
+              "queryString": "CREATE TABLE SUM_PER_SOURCE WITH (KAFKA_TOPIC='SUM_PER_SOURCE', PARTITIONS=6, REPLICAS=3, VALUE_FORMAT='AVRO') AS SELECT\n  EVENTLOGS.EVENTSOURCEIP ROWKEY,\n  AS_VALUE(EVENTLOGS.EVENTSOURCEIP) EVENTSOURCEIP,\n  SUM(EVENTLOGS.EVENTDURATION) SUM\nFROM EVENTLOGS EVENTLOGS\nWHERE (EVENTLOGS.RESULT = 'Pass')\nGROUP BY EVENTLOGS.EVENTSOURCEIP\nEMIT CHANGES;",
```
Comment (ybyzek) [inline cloud-etl/docs/index.rst:378]: Does this need to be updated as well?  `SUM(EVENTDURATION) as sum`

Comment (ybyzek) [reply cloud-etl/docs/index.rst:378]: (same for other queryString too).

i.e., perhaps you updated `statements.sql` after this, so this is still the old query without `as`

Comment (rspurgeon) [reply cloud-etl/docs/index.rst:378]: I just double checked, this is what's rendered by ksqlDB from:
`
CREATE TABLE sum_per_source WITH (KAFKA_TOPIC='SUM_PER_SOURCE', PARTITIONS=6, VALUE_FORMAT='AVRO') AS SELECT eventSourceIP as ROWKEY, as_value(eventSourceIP) as eventSourceIP, SUM(EVENTDURATION) as sum FROM eventlogs WHERE (RESULT = 'Pass') GROUP BY eventSourceIP EMIT CHANGES;
`

**Pull request merged**